Makefile

```diff
@@ -6,9 +6,9 @@ GOMOBILE=$(GOBIN)/gomobile
 GOBIND=env PATH="$(GOBIN):$(PATH)" "$(GOMOBILE)" bind
 IMPORT_PATH=github.com/Jigsaw-Code/outline-go-tun2socks
 
-.PHONY: android apple linux windows intra clean clean-all
+.PHONY: android apple apple_catalyst linux windows intra clean clean-all
```

Comment:
In the interest of consistency, could you rename the target and directory to `mac_catalyst`? that's how Apple calls it: https://developer.apple.com/mac-catalyst/

Reply:
We could also call it something like `apple_future`, since it's intended to replace the `apple` target.  Alternatively, we could rename the other target to `apple_legacy`.

Reply:
apple_mac_catalyst is ok, though I think apple is redundant in that case. I'm fine either way.

apple_future/legacy: I think it's helpful to convey the idea that we are experimenting with the mac_catalyst on the new binary.